src/module/canvas/token/object.ts

```diff
@@ -218,6 +219,17 @@ class TokenPF2e extends Token<TokenDocumentPF2e> {
     async showFloatyText(params: number | ShowFloatyEffectParams): Promise<void> {
         if (!this.isVisible) return;
 
+        /**
+         * If the floaty text is generated by an effect being created/deleted
+         * We do not display it if the effect is unidentified
+         */
+        if (!game.user.isGM && typeof params !== "number") {
+            const [change, document] = Object.entries(params)[0];
+            if (document.type === "effect" && (change === "create" || change === "delete")) {
+                if ((document as EffectPF2e).system.unidentified) return;
+            }
```

Comment:
Get params object and check for `"create" in effectParams`. If `document` is a document, type it as that rather than asserting it.